tests/test_filter.py

```diff
@@ -408,6 +430,51 @@ def test_application_to_dataset():
         filter.apply(dataset, ["yy", "x"])
 
 
+@pytest.mark.parametrize(
+    "filter_args",
+    [
+        dict(
+            filter_scale=8.0,
+            dx_min=1.0,
+            n_steps=0,
+            filter_shape=FilterShape.GAUSSIAN,
+            n_iterations=1,
+        )
+    ],
+)
+def test_iterated_filter(grid_type_and_input_ds, filter_args):
+    "Test that the iterated Gaussian filter gives a result close to the original"
+
+    grid_type, da, grid_vars = grid_type_and_input_ds
+
+    iterated_filter_args = filter_args.copy()
+    iterated_filter_args["n_iterations"] = 2
+
+    filter = Filter(grid_type=grid_type, grid_vars=grid_vars, **filter_args)
+    iterated_filter = Filter(
+        grid_type=grid_type, grid_vars=grid_vars, **iterated_filter_args
+    )
+
+    filtered = filter.apply(da, dims=["y", "x"])
+    iteratively_filtered = iterated_filter.apply(da, dims=["y", "x"])
+
+    area = 1
+    for k, v in grid_vars.items():
+        if "area" in k:
+            area = v
+            break
+
+    # The following tests whether a relative error bound in L^2 holds.
+    # If the polynomial approximations are accurate to within 0.01,
+    # then the precise bound would have 0.0301 ** 2 instead of 0.04 ** 2.
+    # See the "Factoring the Gaussian Filter" section of the docs for details.
+    # Bumped up the tolerance because the default n_steps is not
+    # guaranteed to give error less than *exactly* 0.01; just quite close.
+    assert (((filtered - iteratively_filtered) ** 2) * area).sum() < (0.04 ** 2) * (
+        (filtered ** 2) * area
+    ).sum()
+
```

Comment:
In `test_iterated_viscosity_filter`, we test against `error_bound * unfiltered` (rather than `error_bound * filtered`). Maybe good to be consistent between the two tests?

Reply:
This was a mistake; it should compare to unfiltered, per the docs. Will fix in the next commit.